rosbag2_transport/src/rosbag2_transport/rosbag2_transport.cpp

```diff
@@ -53,7 +53,7 @@ Player::~Player()
 void Player::play(
   const rosbag2_storage::StorageOptions & storage_options, const PlayOptions & play_options)
 {
-  auto player = std::make_shared<impl::Player>(reader_, storage_options, play_options);
+  auto player = std::make_shared<impl::Player>(std::move(reader_), storage_options, play_options);
```

Comment:
Does this mean that you can't call `play` twice, because it has lost ownership of the reader, which is destroyed at the end of this scope?

Reply:
you can't call `rosbag2_transport::play` twice with the current implementation without re-opening the storage. The goal here is still to eventually call `play` on the `Player`. As long as the player instance is active, the storage is attached.

In a follow-up PR I am planning on introducing a `Player::release` function, which - just like a unique_ptr - releases ownership over the storage.

Reply:
Ok - we can move forward - I know it's not actually used that way anywhere - just that it is possible to write code against this API as-is that will confusingly have lost ownership over the reader.

Reply:
hmm, you're right though. That's a real API changer here. I'll add the `release` functions on top of this PR to maintain the same behavior when calling `.play` twice in a row on the `rosbag2_transport::Player`.

Reply:
@emersonknapp this should be taken care of now with the modification to the scope exit object below. Essentially, before destroying the `impl::Player` object, we extract the reader from it to re-use it with the next call to `rosbag2_transport::play`. 
While this might not be the most elegant way, it maintains existing behavior in such that I can call `play` twice in a row.
I've added a test for it as well.